<workflow-app xmlns="uri:oozie:workflow:0.4" name="ptm_rdbcleaner">
	
	<parameters>
        <property>
            <name>rdbUrl</name>
            <description>database url</description>
        </property>
        <property>
            <name>rdbUsername</name>
            <description>database user name</description>
        </property>
        <property>
            <name>rdbPassword</name>
            <description>database user password</description>
        </property>
	</parameters>

    <global>
        <job-tracker>${jobTracker}</job-tracker>
        <name-node>${nameNode}</name-node>
        <configuration>
            <property>
                <name>mapreduce.job.queuename</name>
                <value>${queueName}</value>
            </property>
            <property>
                <name>oozie.launcher.mapred.job.queue.name</name>
                <value>${oozieLauncherQueueName}</value>
            </property>
        </configuration>
    </global>
    
    
	<start to="rdb-cleaner" />
	
	<action name="rdb-cleaner">
		<java>
			<main-class>eu.dnetlib.iis.common.java.ProcessWrapper</main-class>
			<arg>eu.dnetlib.iis.wf.ptm.avro2rdb.RdbDataCleaner</arg>
            <!-- set of tables should be aligned with avro2rdb set of tables data is generated into -->
            <arg>-PtableNamesCSV=Publication,Project,PubProject,PubKeyword,PubFulltext,Citation,PubCitation,PubPDBCode</arg>
            <arg>-PrdbUrl=${rdbUrl}</arg>
            <arg>-PrdbUsername=${rdbUsername}</arg>
            <arg>-PrdbPassword=${rdbPassword}</arg>
		</java>
		<ok to="end" />
		<error to="fail" />
	</action>
    
	<kill name="fail">
		<message>Unfortunately, the process failed -- error message:
			[${wf:errorMessage(wf:lastErrorNode())}]</message>
	</kill>
	
    <end name="end" />
    
</workflow-app>
